security-controls/sast.py

```diff
@@ -0,0 +1,311 @@
+"""Update SAST security-controls data."""
+
+import logging
+import os
+import re
+
+import requests
+from google.cloud import firestore
+from requests.auth import HTTPBasicAuth
+
+SECURITY_CONTROLS = os.environ['SC_FIRESTORE_COLLECTION']
+SAST_DETAILS = os.environ['SAST_FIRESTORE_COLLECTION']
+
+SC_PREVIOUS = "security-controls-previous"
+ID = "id"
+SAST = "sast"
+SAST_LINK = "sast_link"
+
+fs = firestore.Client()
+
+CODACY_BASE = "https://app.codacy.com/api/v3"
+CODACY_API_KEY=os.getenv("CODACY_API_KEY").strip()
+CODACY_URL = "https://app.codacy.com/gh/{org}/{repo}/dashboard"
+
+SONARCLOUD_BASE = "https://sonarcloud.io/api"
+SONARCLOUD_API_KEY=os.getenv("SONARCLOUD_API_KEY").strip()
+SONARCLOUD_URL = "https://sonarcloud.io/project/overview?id={project_key}"
+
+GITHUB_GQL_ENDPOINT = "https://api.github.com/graphql"
+GITHUB_TOKEN=os.getenv("GITHUB_TOKEN").strip()
+
+CODACY_HEADERS = {
+    'Accept': 'application/json',
+    'api-token': CODACY_API_KEY
+}
+
+def codacy_org(organization: str) -> str:
+    '''path for codacy org'''
+    return f'{CODACY_BASE}/organizations/gh/{organization}'
+
+
+repo_re = re.compile(r'https://(www.)?github.com/([a-zA-Z0-9\-_]+)/([a-zA-Z0-9\-_]+)')
+RE_GROUP_ORG = 2
+RE_GROUP_REPO = 3
+
+SONAR_ORGS = {
+    "broadinstitute": "dsp-appsec",
+    "DataBiosphere": "broad-databiosphere",
+    "CancerDataAggregator": "cancerdataaggregator"
+}
+
+CODACY_ORGS = [
```

Comment:
Can we set this in cloud build variables, so we don't have hardcoded values in the code.

Reply:
You make a good point about hardcoding. But the org list does not differ by environment and it seems better/simpler/easier to move it to the deploy script.

Reply:
I understand, but since we want this project to be open-sourced, we should avoid hardcoding values.

Reply:
You're right. I think firestore is probably the best place.

Reply:
Added as build parameters.